sycl/test/vitis/simple_tests/loop_annotations.cpp

```diff
@@ -0,0 +1,37 @@
+// REQUIRES: vitis
+
+// RUN: rm -rf %t.dir && mkdir %t.dir && cd %t.dir
+// RUN: %clangxx -std=c++20 -fsycl -fsycl-targets=%sycl_triple %s -o %t.dir/exec.out
+// RUN: %ACC_RUN_PLACEHOLDER %t.dir/exec.out
+
+#include <sycl.hpp>
+#include <sycl/ext/xilinx/fpga.hpp>
+
+namespace xlx = sycl::ext::xilinx;
+
+int main() {
+  sycl::buffer<int> answer{1};
+  sycl::queue q;
+
+  q.submit([&](sycl::handler &cgh) {
+    sycl::accessor a{answer, cgh, sycl::write_only};
+    cgh.single_task<class forty_two>([=] {
+      for (int i = 0; i < a.get_size(); i++)
+        xlx::pipeline([&] { a[i] = 0; });
+      for (int i = 0; i < a.get_size(); i++)
+        xlx::pipeline<>([&] { a[i] = 0; });
+      for (int i = 0; i < a.get_size(); i++)
+        xlx::annot<xlx::pipeline<>>([&] { a[i] = 0; });
+      for (int i = 0; i < a.get_size(); i++)
+        xlx::annot<xlx::pipeline<>, xlx::unroll<>>([&] { a[i] = 0; });
```

Comment:
Perhaps we could investigate alternative notations like
```suggestion
        annotate(xlx::pipeline, xlx::unroll)([&] { a[i] = 0; });
```
with `annotate` found by ADL, `xlx::pipeline` being a variable of an object having also a variadic call operator to add details.
We could also use C++23 variadic `operator[]` to use `[]` instead of `()` ;-)

Reply:
this should be possible.